---
title: "TVE 16s community composition"
author: "Nicola G. Kriefall, updated by Hannah A."
date: "3/12/2024"
output:
 rmarkdown::html_document:
    theme: cerulean
    toc: yes
    toc_float: yes
    highlight: haddock
    number_sections: true
---

# Setup

```{r packages composition}
library(rlang)
library(stringr)
library(dplyr)
library(stats)
library(ggpubr)
library(vegan)
library(cowplot)
library(tidyverse)
#library(MCMC.OTU)
#install.packages("remotes")
#remotes::install_github("Jtrachsel/funfuns")
library(funfuns)
library(phyloseq)
```

Read in data

```{r read in data}
setwd("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability")

load("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/taxa2.Rdata")

#T0 phyloseq objects
load("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/ps.clean.t0.Rdata")
load("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/ps.rare.t0.Rdata")
load("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/ps.trim.rare.t0.Rdata") # rare otu's trimmed, then rarefied (1k)

#Prestress phyloseq objects
load("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/ps.clean.prestress.Rdata")
load("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/ps.rare.prestress.Rdata")
load("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/ps.trim.rare.prestress.Rdata") # rare otu's trimmed, then rarefied (1k)


# remove Control 2, and lineage NA's
ps.clean.t0 <- subset_samples(ps.clean.t0,(!is.na(lineage)))
ps.clean.t0 <- subset_samples(ps.clean.t0,(!is.na(treat)))
ps.clean.t0 <- subset_samples(ps.clean.t0,lineage!="L3")

ps.rare.t0 <- subset_samples(ps.rare.t0,(!is.na(lineage)))
ps.rare.t0 <- subset_samples(ps.rare.t0,(!is.na(treat)))
ps.rare.t0 <- subset_samples(ps.rare.t0,lineage!="L3")

ps.trim.rare.t0 <- subset_samples(ps.trim.rare.t0,(!is.na(lineage)))
ps.trim.rare.t0 <- subset_samples(ps.trim.rare.t0,(!is.na(treat)))
ps.trim.rare.t0 <- subset_samples(ps.trim.rare.t0,lineage!="L3")

# remove Control 2, low and high variability, and lineage NA's
ps.clean.prestress <- subset_samples(ps.clean.prestress,(!is.na(lineage)))
ps.clean.prestress <- subset_samples(ps.clean.prestress,(!is.na(treat)))
ps.clean.prestress <- subset_samples(ps.clean.prestress,(treat!="Control 2"))
ps.clean.prestress <- subset_samples(ps.clean.prestress,(treat!="Low Var"))
ps.clean.prestress <- subset_samples(ps.clean.prestress,(treat!="High Var"))
ps.clean.prestress <- subset_samples(ps.clean.prestress,lineage!="L3")

ps.rare.prestress <- subset_samples(ps.rare.prestress,(!is.na(lineage)))
ps.rare.prestress <- subset_samples(ps.rare.prestress,(!is.na(treat)))
ps.rare.prestress <- subset_samples(ps.rare.prestress,(treat!="Control 2"))
ps.rare.prestress <- subset_samples(ps.rare.prestress,(treat!="Low Var"))
ps.rare.prestress <- subset_samples(ps.rare.prestress,(treat!="High Var"))
ps.rare.prestress <- subset_samples(ps.rare.prestress,lineage!="L3")

ps.trim.rare.prestress <- subset_samples(ps.trim.rare.prestress,(!is.na(lineage)))
ps.trim.rare.prestress <- subset_samples(ps.trim.rare.prestress,(!is.na(treat)))
ps.trim.rare.prestress <- subset_samples(ps.trim.rare.prestress,(treat!="Control 2"))
ps.trim.rare.prestress <- subset_samples(ps.trim.rare.prestress,(treat!="Low Var"))
ps.trim.rare.prestress <- subset_samples(ps.trim.rare.prestress,(treat!="High Var"))
ps.trim.rare.prestress <- subset_samples(ps.trim.rare.prestress,lineage!="L3")

# moving forward with rarefied data, doesn't seem to make a difference in the results whether rare OTUs trimmed or not and want to keep same dataset / # of individuals that we have for the diversity analyses

# make the factors of interest actual factors - T0
sample_data(ps.rare.t0)$sitename = as.factor(sample_data(ps.rare.t0)$sitename)
levels(sample_data(ps.rare.t0)$sitename)

sample_data(ps.rare.t0)$lineage = as.factor(sample_data(ps.rare.t0)$lineage)
levels(sample_data(ps.rare.t0)$lineage)

# make the factors of interest actual factors - PreStress
sample_data(ps.rare.prestress)$sitename = as.factor(sample_data(ps.rare.prestress)$sitename)
levels(sample_data(ps.rare.prestress)$sitename)

sample_data(ps.rare.prestress)$lineage = as.factor(sample_data(ps.rare.prestress)$lineage)
levels(sample_data(ps.rare.prestress)$lineage)

sample_data(ps.rare.prestress)$treat = factor(sample_data(ps.rare.prestress)$treat)
levels(sample_data(ps.rare.prestress)$treat)

# set color palettes
cols_site_diverging <- c("CI" = "#543005", "PD"= "#bf812d",  "SP"= "#dfc27d",  "BN" = "#003c30", "BS"= "#35978f", "CA"= "#80cdc1")
cols_treat_reds <- c("darkgrey", "#CC3300")
cols_lineage <- c("L1" = "#3f007d", "L2" = "#807dba")
```

Rename ASVs to be more informative

```{r rename ASVs}
tax.t0 <- as.data.frame(ps.rare.t0@tax_table@.Data)

tax.clean.t0 <- data.frame(row.names = row.names(tax.t0),
                        Kingdom = str_replace(tax.t0[,1], "D_0__",""),
                        Phylum = str_replace(tax.t0[,2], "D_1__",""),
                        Class = str_replace(tax.t0[,3], "D_2__",""),
                        Order = str_replace(tax.t0[,4], "D_3__",""),
                        Family = str_replace(tax.t0[,5], "D_4__",""),
                        Genus = str_replace(tax.t0[,6], "D_5__",""),
                        Species = str_replace(tax.t0[,7], "D_6__",""),
                        stringsAsFactors = FALSE)
tax.clean.t0[is.na(tax.clean.t0)] <- ""

for (i in 1:7){ tax.clean.t0[,i] <- as.character(tax.clean.t0[,i])}
####### Fill holes in the tax table
tax.clean.t0[is.na(tax.clean.t0)] <- ""
for (i in 1:nrow(tax.clean.t0)){
  if (tax.clean.t0[i,2] == ""){
    kingdom <- paste("Kingdom_", tax.clean.t0[i,1], sep = "")
    tax.clean.t0[i, 2:7] <- kingdom
  } else if (tax.clean.t0[i,3] == ""){
    phylum <- paste("Phylum_", tax.clean.t0[i,2], sep = "")
    tax.clean.t0[i, 3:7] <- phylum
  } else if (tax.clean.t0[i,4] == ""){
    class <- paste("Class_", tax.clean.t0[i,3], sep = "")
    tax.clean.t0[i, 4:7] <- class
  } else if (tax.clean.t0[i,5] == ""){
    order <- paste("Order_", tax.clean.t0[i,4], sep = "")
    tax.clean.t0[i, 5:7] <- order
  } else if (tax.clean.t0[i,6] == ""){
    family <- paste("Family_", tax.clean.t0[i,5], sep = "")
    tax.clean.t0[i, 6:7] <- family
  } else if (tax.clean.t0[i,7] == ""){
    tax.clean.t0$Species[i] <- paste("Genus",tax.clean.t0$Genus[i], sep = "_")
  }
}

tax_table(ps.rare.t0) <- as.matrix(tax.clean.t0)

# now prestress
tax.ps <- as.data.frame(ps.rare.prestress@tax_table@.Data)

tax.clean.ps <- data.frame(row.names = row.names(tax.ps),
                        Kingdom = str_replace(tax.ps[,1], "D_0__",""),
                        Phylum = str_replace(tax.ps[,2], "D_1__",""),
                        Class = str_replace(tax.ps[,3], "D_2__",""),
                        Order = str_replace(tax.ps[,4], "D_3__",""),
                        Family = str_replace(tax.ps[,5], "D_4__",""),
                        Genus = str_replace(tax.ps[,6], "D_5__",""),
                        Species = str_replace(tax.ps[,7], "D_6__",""),
                        stringsAsFactors = FALSE)
tax.clean.ps[is.na(tax.clean.ps)] <- ""

for (i in 1:7){ tax.clean.ps[,i] <- as.character(tax.clean.ps[,i])}
####### Fill holes in the tax table
tax.clean.ps[is.na(tax.clean.ps)] <- ""
for (i in 1:nrow(tax.clean.ps)){
  if (tax.clean.ps[i,2] == ""){
    kingdom <- paste("Kingdom_", tax.clean.ps[i,1], sep = "")
    tax.clean.ps[i, 2:7] <- kingdom
  } else if (tax.clean.ps[i,3] == ""){
    phylum <- paste("Phylum_", tax.clean.ps[i,2], sep = "")
    tax.clean.ps[i, 3:7] <- phylum
  } else if (tax.clean.ps[i,4] == ""){
    class <- paste("Class_", tax.clean.ps[i,3], sep = "")
    tax.clean.ps[i, 4:7] <- class
  } else if (tax.clean.ps[i,5] == ""){
    order <- paste("Order_", tax.clean.ps[i,4], sep = "")
    tax.clean.ps[i, 5:7] <- order
  } else if (tax.clean.ps[i,6] == ""){
    family <- paste("Family_", tax.clean.ps[i,5], sep = "")
    tax.clean.ps[i, 6:7] <- family
  } else if (tax.clean.ps[i,7] == ""){
    tax.clean.ps$Species[i] <- paste("Genus",tax.clean.ps$Genus[i], sep = "_")
  }
}

tax_table(ps.rare.prestress) <- as.matrix(tax.clean.ps)
```


# All ASVs - PCAs

## Rarefied Data

```{r site and lineage T0 - rarefied}
ord.t0 <- ordinate(ps.rare.t0, "PCoA", "bray")

pcoa.lineage.t0 = plot_ordination(ps.rare.t0,ord.t0,color="lineage",shape="sitename")+
  geom_point(size = 2.5, alpha=0.8)+
  scale_color_manual(name="Lineage",values=cols_lineage)+
  stat_ellipse(aes(group=ps.rare.t0@sam_data$lineage), type = "t", lwd = 1)+
  scale_shape_manual(name="Site", values = c(15,16,17,22,21,24))+
  theme_bw()
pcoa.lineage.t0
ggsave(pcoa.lineage.t0, filename = "pcoas.allasvs.rare.lineage.t0.pdf", width=5, height=4, units=c("in"), useDingbats=FALSE)

pcoa.site.t0 = plot_ordination(ps.rare.t0, ord.t0, color="sitename", shape="sitename")+
  geom_point(alpha=0.8)+
  scale_color_manual(name="Site",values=cols_site_diverging)+
  scale_shape_manual(name="Site",values=c(15,16,17,22,21,24))+
  stat_ellipse()+
  theme_bw()
pcoa.site.t0
ggsave(pcoa.site.t0, filename = "pcoas.allasvs.rare.site.t0.pdf", width=4, height=3, units=c("in"), useDingbats=FALSE)

```


```{r site, treatment and lineage prestress - rarefied}

# now prestress data rarefied
ord.ps <- ordinate(ps.rare.prestress, "PCoA", "bray")

pcoa.site.ps = plot_ordination(ps.rare.prestress, ord.ps, color="sitename", shape="sitename")+
  geom_point(alpha=0.8)+
  scale_color_manual(name="Site",values=cols_site_diverging)+
  scale_shape_manual(name="Site",values=c(15,16,17,22,21,24))+
  stat_ellipse()+
  theme_bw()

pcoa.lineage.ps = plot_ordination(ps.rare.prestress,ord.ps,color="lineage")+
  geom_point(alpha=0.8)+
  scale_color_manual(name="Lineage",values=cols_lineage)+
  stat_ellipse()+
  theme_bw()

pcoa.treat.ps = plot_ordination(ps.rare.prestress,ord.ps, color="treat", shape="treat")+
  geom_point(alpha=0.8)+
  scale_color_manual(name="Treatment",values=cols_treat_reds)+
  scale_shape_manual(name="Treatment",values=c(15,16))+
  stat_ellipse()+
  theme_bw()

gg.pcoa.all.rare.ps = ggarrange(pcoa.lineage.ps, pcoa.site.ps, pcoa.treat.ps,
                                  labels = c("(a)","(b)","(c)"),
                                  nrow=1,
                                  common.legend=F,legend="none")

ggsave(gg.pcoa.all.rare.ps, filename = "pcoas.alldata.rare.prestress.pdf", width=8, height=3, units=c("in"), useDingbats=FALSE)

```


### Rarefied Stats

Help on adonis (here)[https://thebiobucket.blogspot.com/2011/04/assumptions-for-permanova-with-adonis.html#more]

```{r stats rarefied data}
seq.rare.t0 <- data.frame(otu_table(ps.rare.t0))

# make distance matrix with vegdist():
dist.rare.t0 <- vegdist(seq.rare.t0)
samdf.rare.t0 <- data.frame(sample_data(ps.rare.t0))
row.names(samdf.rare.t0)==row.names(seq.rare.t0)

bet.all.t0 <- betadisper(dist.rare.t0,samdf.rare.t0$sitename)
anova(bet.all.t0) 
#           Df   Sum Sq   Mean Sq F value Pr(>F)
# Groups     5 0.005796 0.0011592  0.4494 0.8112
# Residuals 41 0.105758 0.0025795               

permutest(bet.all.t0, pairwise = TRUE, permutations = 999) # all ns
plot(bet.all.t0) 

bet.all.t0 <- betadisper(dist.rare.t0,samdf.rare.t0$lineage)
anova(bet.all.t0) #p=0.851
plot(bet.all.t0) 

adonis2(seq.rare.t0 ~ lineage, data=samdf.rare.t0, permutations=999)
#          Df SumOfSqs      R2      F Pr(>F)   
# lineage   1   0.5992 0.03147 1.4621  0.003 **
# Residual 45  18.4432 0.96853                 
# Total    46  19.0424 1.00000                 

adonis2(seq.rare.t0 ~ sitename, data=samdf.rare.t0, permutations=999)
#          Df SumOfSqs      R2      F Pr(>F)    
# sitename  5   2.7388 0.14383 1.3775  0.001 ***
# Residual 41  16.3036 0.85617                  
# Total    46  19.0424 1.00000                  

# now prestress data
seq.rare.ps <- data.frame(otu_table(ps.rare.prestress))

# make distance matrix with vegdist():
dist.rare.ps <- vegdist(seq.rare.ps)
samdf.rare.ps <- data.frame(sample_data(ps.rare.prestress))
row.names(samdf.rare.ps)==row.names(seq.rare.ps)

bet.all.ps <- betadisper(dist.rare.ps,samdf.rare.ps$sitename)
anova(bet.all.ps) 
#           Df   Sum Sq   Mean Sq F value Pr(>F)
# Groups     5 0.014839 0.0029678  1.1194 0.3593
# Residuals 63 0.167022 0.0026511               

plot(bet.all.ps) 

bet.all.ps <- betadisper(dist.rare.ps,samdf.rare.ps$lineage)
anova(bet.all.ps)
#           Df   Sum Sq   Mean Sq F value  Pr(>F)  
# Groups     1 0.006205 0.0062050  4.5712 0.03616 *
# Residuals 67 0.090947 0.0013574                  

plot(bet.all.ps) 

bet.all.ps <- betadisper(dist.rare.ps,samdf.rare.ps$treat)
anova(bet.all.ps)
#           Df   Sum Sq    Mean Sq F value Pr(>F)
# Groups     1 0.000552 0.00055217  0.3525 0.5547
# Residuals 67 0.104959 0.00156656               

plot(bet.all.ps) 

adonis2(seq.rare.ps ~ treat+lineage+sitename, data=samdf.rare.ps, permutations=999)
#          Df SumOfSqs      R2      F Pr(>F)  
# treat     1   0.6103 0.02043 1.4151  0.034 *
# lineage   1   0.5445 0.01823 1.2625  0.097 .
# sitename  5   2.4113 0.08071 1.1182  0.115  
# Residual 61  26.3079 0.88063                

```


# Hannah stopped here for revision. Below is script to determine core and accessory microbiomes for the same corals.

# Core vs. accessory

## Core Microbiome

```{r packages core, echo=FALSE}
#BiocManager::install("microbiome")
#remotes::install_github("r-lib/rlang")
library(microbiome)
```

```{r core}
# prevalence here means what percent of samples does the taxa need to be in to be considered 'core' 

# Transform to compositional abundances
pseq.rel.t0 <- microbiome::transform(ps.rare.t0, "compositional")
pseq.rel.ps <- microbiome::transform(ps.rare.prestress, "compositional")

# Calculate prevalences for all taxonomic groups
head(prevalence(ps.rare.t0, detection = 1/100, sort = TRUE))
head(prevalence(ps.rare.prestress, detection = 1/100, sort = TRUE))

# Pick the core (>0.1% relative abundance in >50% of the samples)
pseq.core.t0 <- core(ps.rare.t0, detection = 0.1/100, prevalence = 0.5)
pseq.core.t0
#27 taxa and 47 samples
pseq.core.ps <- core(ps.rare.prestress, detection = 0.1/100, prevalence = 0.5)
pseq.core.ps
#4 taxa and 69 samples

#saving
core.tax.t0 <- data.frame(pseq.core.t0@tax_table)
core.tax.ps <- data.frame(pseq.core.ps@tax_table)

# Plot bar plots
# t0
ps_glom <- tax_glom(pseq.core.t0, "Genus")
ps0 <- transform_sample_counts(ps_glom, function(x) x / sum(x))
ps1 <- merge_samples(ps0, "lineage")
ps2 <- transform_sample_counts(ps1, function(x) x / sum(x))

plot.t0 = plot_bar(ps2, fill="Genus")+
  geom_bar(stat="identity")+
  theme_bw()
  #scale_fill_brewer(palette="BrBG")
plot.t0
#ggsave(plot, file="rare.t0.core.bar.pdf",width=8)

# prestress
ps_glom <- tax_glom(pseq.core.ps, "Genus")
ps0 <- transform_sample_counts(ps_glom, function(x) x / sum(x))
ps1 <- merge_samples(ps0, "lineage")
ps2 <- transform_sample_counts(ps1, function(x) x / sum(x))

plot.ps = plot_bar(ps2, fill="Genus")+
  geom_bar(stat="identity")+
  theme_bw()
  #scale_fill_brewer(palette="BrBG")
plot.ps
#ggsave(plot, file="rare.t0.core.bar.pdf",width=8)

# Core Bray-Curtis PCoAs - T0
#not rel abun
plot.core.site.t0 = plot_ordination(pseq.core.t0,ordinate(pseq.core.t0,"PCoA", "bray"),color="sitename",shape="sitename")+
  stat_ellipse()+
  theme_bw()+
  scale_color_manual(name = "Site", 
                     values=cols_site_diverging,
                     breaks = c("BN","BS","CA","CI","PD","SP"))+
  scale_shape_manual(name = "Site", 
                     values=c(15,16,17,22,21,24),
                     breaks = c("BN","BS","CA","CI","PD","SP"))
plot.core.site.t0

# lineage pcoa's 
plot.core.lin.t0 = plot_ordination(pseq.core.t0,ordinate(pseq.core.t0,"PCoA", "bray"),color="lineage")+
  stat_ellipse()+
  theme_bw()+
  scale_color_manual(name = "Lineage", 
                     values=cols_lineage,
                     breaks = c("L1","L2"))
plot.core.lin.t0

plots.core.t0 = ggarrange(plot.core.site.t0, plot.core.lin.t0, nrow = 1, ncol = 2, legend = "none")
ggsave(plots.core.t0, filename = "pcoas.core.rare.t0.pdf", width=6, height = 3, units=c("in"), useDingbats=FALSE)


#rel abundance
pseq.core.rel.t0 <- transform_sample_counts(pseq.core.t0, function(x) x / sum(x))
# sitename pcoa's 
plot.core.rel.site.t0 = plot_ordination(pseq.core.rel.t0,ordinate(pseq.core.rel.t0,"PCoA", "bray"),color="sitename",shape="sitename")+
  stat_ellipse()+
  theme_bw()+
  scale_color_manual(name = "Site", 
                     values=cols_site_diverging,
                     breaks = c("BN","BS","CA","CI","PD","SP"))+
  scale_shape_manual(name = "Site", 
                     values=c(15,16,17,22,21,24),
                     breaks = c("BN","BS","CA","CI","PD","SP"))
plot.core.rel.site.t0

# lineage pcoa's 
plot.core.rel.lin.t0 = plot_ordination(pseq.core.rel.t0,ordinate(pseq.core.rel.t0,"PCoA", "bray"),color="lineage")+
  stat_ellipse()+
  theme_bw()+
  scale_color_manual(name = "Lineage", 
                     values=cols_lineage,
                     breaks = c("L1","L2"))
plot.core.rel.lin.t0

plots.core.rel.t0 = ggarrange(plot.core.rel.site.t0, plot.core.rel.lin.t0, nrow = 1, ncol = 2, legend = "none")
ggsave(plots.core.rel.t0, filename = "pcoas.core.rel.rare.t0.pdf", width=6, height = 3, units=c("in"), useDingbats=FALSE)


# Core Bray-Curtis PCoAs - PreStress - not enough taxa to make these plots
```


```{r calculate core abundance}
# calculating core abundances #
core.sqs.t0 <- tax_table(pseq.core.t0)
core.sqs.ids.t0 <- row.names(core.sqs.t0)
core.sqs.ids.t0

ps.rare.t0.rel <- transform_sample_counts(ps.rare.t0, function(x) x / sum(x))
seq.rare.rel.t0 <- data.frame(otu_table(ps.rare.t0.rel))
tax.core.t0 <- tax_table(ps.rare.t0.rel)

seq.core.t0 <- seq.rare.rel.t0 %>% select(all_of(core.sqs.ids.t0))
core.rel.t0 <- data.frame(colMeans(seq.core.t0))

total.rel.t0 <- data.frame(colMeans(seq.rare.rel.t0))
total.rel.ordered.t0 <- data.frame(total.rel.t0[order(-total.rel.t0$colMeans.seq.rare.rel.t0.),,drop=FALSE])

samdf.core.t0 <- data.frame(sample_data(pseq.core.t0))
```

### Core stats

```{r core stats Lineages together}
# not doing prestress data because we only have 4 taxa
seq.core <- data.frame(otu_table(pseq.core.t0))

dist.core <- vegdist(seq.core)
samdf.core <- data.frame(sample_data(pseq.core.t0))
row.names(samdf.core)==row.names(seq.core)

# site
bet.all <- betadisper(dist.core,samdf.core$sitename)
anova(bet.all) # p=0.2995
permutest(bet.all, pairwise = TRUE, permutations = 999)
#          BN       BS       CA       CI       PD    SP
# BN          0.226000 0.247000 0.154000 0.025000 0.015
# BS 0.218486          0.974000 0.835000 0.286000 0.536
# CA 0.240361 0.975253          0.805000 0.271000 0.528
# CI 0.135811 0.839872 0.821022          0.363000 0.648
# PD 0.011451 0.273823 0.275007 0.371321          0.594
# SP 0.014493 0.522893 0.510038 0.658436 0.599421      

plot(bet.all) 

# lineage
bet.all <- betadisper(dist.core,samdf.core$lineage)
anova(bet.all) # p=0.3628
#           Df  Sum Sq  Mean Sq F value Pr(>F)
# Groups     1 0.01336 0.013355  0.8454 0.3628
# Residuals 45 0.71091 0.015798               

permutest(bet.all, pairwise = TRUE, permutations = 999) #all ns
plot(bet.all) 

#adonis(seq.core ~ sitename, data=samdf.core, permutations=999) #p=0.233
adonis2(seq.core ~ lineage, data=samdf.core, permutations=999) 
#          Df SumOfSqs      R2     F Pr(>F)
# lineage   1   0.2821 0.02629 1.215  0.265
# Residual 45  10.4479 0.97371             
# Total    46  10.7300 1.00000             

```


## Core Summary

When analyzing lineages together, nothing significantly affects dispersion or adonis (not treatment, lineage, or site of origin). Essentially, the core microbiome is very resistant to changes regardless of the factor of interest.

When analyzing the core microbiome separately for Lineage 1 + 2, the only thing significant is effect of treatment on Lineage 2 - and this is marginally significant (p = 0.05).

## Accessory

```{r accessory lineages together}
# not doing prestress data here because we only have 4 taxa for prestress data
ps.rare.t0.otu <- data.frame(ps.rare.t0@otu_table)
core.tax <- data.frame(pseq.core.t0@tax_table)
core.ids <- c(rownames(core.tax))
ps.rare.t0.acc.otu <- ps.rare.t0.otu[,!colnames(ps.rare.t0.otu) %in% core.ids ]
row.names(samdf.core.t0) <- samdf.core.t0$frag

#remake phyloseq object
ps.acc <- phyloseq(otu_table(ps.rare.t0.acc.otu, taxa_are_rows=FALSE), 
                         sample_data(samdf.core.t0), 
                         tax_table(taxa2))
ps.acc #6117 taxa and 47 samples


# make bray-curtis dissimilarity pcoas for accessory microbiome

#sitename
pcoa.acc.all.site.t0 = plot_ordination(ps.acc,ordinate(ps.acc,"PCoA", "bray"),color="sitename", shape="sitename")+
  geom_point(alpha=0.8)+
  scale_color_manual(name="Site",
                     values=cols_site_diverging,
                     breaks = c("BN","BS","CA","CI","PD","SP"))+
  scale_shape_manual(name="Site",
                     values=c(15,16,17,22,21,24),
                     breaks = c("BN","BS","CA","CI","PD","SP"))+
  stat_ellipse()+
  theme_bw()
pcoa.acc.all.site.t0

#lineage
pcoa.acc.all.lin.t0 = plot_ordination(ps.acc,ordinate(ps.acc,"PCoA", "bray"),color="lineage")+
  geom_point(alpha=0.8)+
  scale_color_manual(name="Lineage",
                     values=cols_lineage,
                     breaks = c("L1","L2"))+
  stat_ellipse()+
  theme_bw()
pcoa.acc.all.lin.t0

gg.pcoa.acc.all.t0 = ggarrange(pcoa.acc.all.site.t0,pcoa.acc.all.lin.t0, nrow=1, ncol=2, common.legend=F,legend="none")
ggsave(gg.pcoa.acc.all.t0, filename = "pcoas.accessory.rare.t0.pdf", width=8, height=4, units=c("in"), useDingbats=FALSE)

#rel abundance
pseq.acc.rel <- transform_sample_counts(ps.acc, function(x) x / sum(x))

#sitename
pcoa.acc.all.rel.site.t0 = plot_ordination(pseq.acc.rel,ordinate(pseq.acc.rel,"PCoA", "bray"),color="sitename", shape="sitename")+
  geom_point(alpha=0.8)+
  scale_color_manual(name="Site",
                     values=cols_site_diverging,
                     breaks = c("BN","BS","CA","CI","PD","SP"))+
  scale_shape_manual(name="Site",
                     values=c(15,16,17,22,21,24),
                     breaks = c("BN","BS","CA","CI","PD","SP"))+
  stat_ellipse()+
  theme_bw()

#lineage
pcoa.acc.all.rel.lin.t0 = plot_ordination(pseq.acc.rel,ordinate(pseq.acc.rel,"PCoA", "bray"),color="lineage")+
  geom_point(alpha=0.8)+
  scale_color_manual(name="Lineage",
                     values=cols_lineage,
                     breaks = c("L1","L2"))+
  stat_ellipse()+
  theme_bw()

gg.pcoa.acc.rel.t0 = ggarrange(pcoa.acc.all.rel.site.t0,pcoa.acc.all.rel.lin.t0, nrow=1, ncol=2, common.legend=F,legend="none")
ggsave(gg.pcoa.acc.rel.t0, filename = "pcoas.accessory.rel.rare.t0.pdf", width=8, height=4, units=c("in"), useDingbats=FALSE)
```



```{r stats acc - lineages together}
seq.acc <- data.frame(otu_table(ps.acc))

dist.acc <- vegdist(seq.acc)
samdf.acc <- data.frame(sample_data(ps.acc))
row.names(samdf.acc)==row.names(seq.acc)

#sitename
bet.all <- betadisper(dist.acc,samdf.acc$sitename)
anova(bet.all) #p=0.4335
permutest(bet.all, pairwise = TRUE, permutations = 999) #CA + BS diff
plot(bet.all) 

#lineage
bet.all <- betadisper(dist.acc,samdf.acc$lineage)
anova(bet.all) #no diff
#           Df   Sum Sq    Mean Sq F value Pr(>F)
# Groups     1 0.001270 0.00126978   1.726 0.1956
# Residuals 45 0.033105 0.00073567               

permutest(bet.all, pairwise = TRUE, permutations = 999) 
plot(bet.all) 

adonis2(seq.acc ~ lineage, data=samdf.acc, permutations=999) 
#          Df SumOfSqs      R2      F Pr(>F)    
# lineage   1    0.676 0.03214 1.4943  0.001 ***
# Residual 45   20.358 0.96786                  
# Total    46   21.034 1.00000                  

```



## Accessory Summary

Both lineage and treatment have a significant effect from adonis when lineages analyzed together. Lineage also has significant difference in dispersion when analyzed together. 

No differences in dispersion by treatment or site (for either lineage) when lineages analyzed separately. Treatment and site are still significantly different from adonis when lineages analyzed separate, but only for L1 - no differences for L2.

# Bar plots {.tabset}

## All, by Phylum

```{r bar plot, eval=FALSE}
#rel abundance
ps.rare.t0.rel <- transform_sample_counts(ps.rare.t0, function(x) x / sum(x))

ps.all.tab.t0 <- psmelt(ps.rare.t0.rel)%>%
  filter(!is.na(Abundance))%>%
  filter(!is.na(Phylum))%>%
  #filter(!is.na(Genus))%>%
  group_by(sitename,treat,lineage,Phylum,OTU)%>%
  #group_by(sitename,lineage,Genus,OTU)%>%
  summarize_at("Abundance",mean)
  #filter(Abundance > .1)


gg.bar.all.phy.t0 <- ggplot(ps.all.tab.t0,aes(x=lineage,y=Abundance,fill=Phylum))+
  geom_bar(stat="identity")+
  theme_bw()+
  #theme(legend.position="none")+
  xlab('Lineage')+  
  facet_wrap(~sitename)
gg.bar.all.phy.t0

ggsave(gg.bar.all.phy.t0, filename = "rare.t0.barplot.all.phylum.pdf", width=10, height=6, units=c("in"), useDingbats=FALSE)

# prestress data
ps.rare.ps.rel <- transform_sample_counts(ps.rare.prestress, function(x) x / sum(x))

ps.all.tab.ps <- psmelt(ps.rare.ps.rel)%>%
  filter(!is.na(Abundance))%>%
  filter(!is.na(Phylum))%>%
  #filter(!is.na(Genus))%>%
  group_by(sitename,treat,lineage,Phylum,OTU)%>%
  #group_by(sitename,lineage,Genus,OTU)%>%
  summarize_at("Abundance",mean)
  #filter(Abundance > .1)


gg.bar.all.phy.ps <- ggplot(ps.all.tab.ps,aes(x=lineage,y=Abundance,fill=Phylum))+
  geom_bar(stat="identity")+
  theme_bw()+
  #theme(legend.position="none")+
  xlab('Lineage')+  
  facet_wrap(~sitename)
gg.bar.all.phy.ps

ggsave(gg.bar.all.phy.ps, filename = "rare.prestress.barplot.all.phylum.pdf", width=10, height=6, units=c("in"), useDingbats=FALSE)

```




## Trimmed + Rarefied data (skipping since we are just using rarefied data)

```{r sites rel abun}
ps.trim.rare.t0.rel <- transform_sample_counts(ps.trim.rare.t0, function(x) x / sum(x))
ord.rel <- ordinate(ps.trim.rare.t0, "PCoA", "bray")

# pcoa.treat.rel = plot_ordination(ps.trim.rel, ord.rel, color="treat", shape="treat")+
#   geom_point(alpha=0.8)+
#   scale_color_manual(name="Treatment",values=cols_treat_reds)+
#   scale_shape_manual(name="Treatment",values=c(15,16,17,18))+
#   stat_ellipse()+
#   theme_bw()

pcoa.site.rel = plot_ordination(ps.trim.rare.t0.rel, ord.rel, color="sitename", shape="sitename")+
  geom_point(alpha=0.8)+
  scale_color_manual(name="Site",values=cols_site_diverging)+
  scale_shape_manual(name="Site",values=c(15,16,17,22,21,24))+
  stat_ellipse()+
  theme_bw()

pcoa.lineage.rel = plot_ordination(ps.trim.rare.t0.rel,ord.rel, color="lineage")+
  geom_point(alpha=0.8)+
  scale_color_manual(name="Lineage",values=cols_lineage)+
  #scale_shape_manual(name="Lineage", values = c(19,17))+
  stat_ellipse()+
  theme_bw()

gg.pcoa.trim.rare.rel = ggarrange(pcoa.site.rel,pcoa.lineage.rel,nrow=1,common.legend=F,legend="none")
ggsave(gg.pcoa.trim.rare.rel, filename = "pcoas.alldata.trim.rare.t0.rel.pdf", width=6, height=3, units=c("in"), useDingbats=FALSE)


#try faceting a few ways to see if I am missing anything

pcoa.treat.rel.sitefacet = plot_ordination(ps.trim.rel, ord.rel, color="treat", shape="treat")+
  geom_point(alpha=0.8)+
  scale_color_manual(name="Treatment",values=cols_treat_reds)+
  scale_shape_manual(name="Treatment",values=c(15,16,17,18))+
  stat_ellipse()+
  theme_bw()+
  facet_wrap(~sitename)

pcoa.lineage.rel.facet = plot_ordination(ps.trim.rel,ord.rel, color="lineage", shape="lineage")+
  geom_point(alpha=0.8)+
  scale_color_manual(name="Lineage",values=cols_lineage)+
  scale_shape_manual(name="Lineage", values = c(16,16))+
  stat_ellipse()+
  theme_bw()+
  facet_wrap(~treat)

gg.pcoa.rel.facets = ggarrange(pcoa.treat.rel.linfacet,pcoa.treat.rel.sitefacet,pcoa.lineage.rel.facet,nrow=1,common.legend=F,legend="none")

ggsave(gg.pcoa.rel.facets, filename = "/Users/hannahaichelman/Documents/BU/TVE/16S_ITS2/16S_PreStress/CommunityComposition/bray_pcoas_all.trim.rel.facets.pdf", width=12, height=6, units=c("in"), useDingbats=FALSE)
```


### Trimmed + Rarefied Stats

```{r stats site rel}
seq.trim <- data.frame(otu_table(ps.trim.rel))

dist.trim <- vegdist(seq.trim)
samdf.trim <- data.frame(sample_data(ps.trim))
row.names(samdf.trim)==row.names(seq.trim)

bet.all <- betadisper(dist.trim,samdf.trim$treat)
anova(bet.all) #p=0.7516
permutest(bet.all, pairwise = TRUE, permutations = 999) # all ns
plot(bet.all) #very much overlap, not sig

bet.all <- betadisper(dist.trim,samdf.trim$sitename)
anova(bet.all) #p=0.9023
permutest(bet.all, pairwise = TRUE, permutations = 999) # all ns
plot(bet.all) #overlap

bet.all <- betadisper(dist.trim,samdf.trim$lineage)
anova(bet.all) #p=0.3134
plot(bet.all) #overlap

adonis(seq.trim ~ treat+lineage, data=samdf.trim, permutations=999) 
#            Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)  
# treat       3     1.267 0.42219  1.7197 0.03289  0.025 *
# lineage     1     0.418 0.41833  1.7040 0.01086  0.088 .
# Residuals 150    36.826 0.24550         0.95625         
# Total     154    38.511                 1.00000         

pairwise.adonis(seq.trim, factors = samdf.trim$treat, permutations = 999)
# same as above - control different from all variabilities
#                 pairs   F.Model         R2 p.value p.adjusted
# 1  Control vs Low Var 2.1567807 0.02948162   0.027      0.027
# 2  Control vs Mod Var 2.4471822 0.02932612   0.014      0.014
# 3 Control vs High Var 2.8835101 0.03799917   0.012      0.012
# 4  Low Var vs Mod Var 0.8824621 0.01118705   0.478      0.478
# 5 Low Var vs High Var 0.7772710 0.01098193   0.557      0.557
# 6 Mod Var vs High Var 1.0663479 0.01315401   0.279      0.279

```

## Plot - both versions of PCoA - rarefied and relative abundance

```{r ggarrange pcoas}
ggarrange(gg.pcoa.all.trim.rare,gg.pcoa.rel,labels="AUTO",common.legend=TRUE,legend="right")

# make final microbiome combined pcoa fig - treatment pca's

pcoas.combined = ggarrange(pcoa.treat.rel, plot.core.rel.treat, pcoa.acc.all.rel.treat,
          labels = c("(a) All ASVs", "(b) Core ASVs", "(c) Accessory ASVs"),
          common.legend = TRUE, legend = "right",
          ncol = 3, nrow = 1)

ggsave(pcoas.combined, filename = "/Users/hannahaichelman/Documents/BU/TVE/16S_ITS2/16S_PreStress/CommunityComposition/all_pcoas_treat.pdf", width=9, height=4, units=c("in"), useDingbats=FALSE)

# make final microbiome combined pcoa fig - lineage pca's

pcoas.combined.lin = ggarrange(pcoa.lineage.rel, plot.core.rel.lin, pcoa.acc.all.rel.lin,
          labels = c("(a) All ASVs", "(b) Core ASVs", "(c) Accessory ASVs"),
          common.legend = TRUE, legend = "right",
          ncol = 3, nrow = 1)

ggsave(pcoas.combined.lin, filename = "/Users/hannahaichelman/Documents/BU/TVE/16S_ITS2/16S_PreStress/CommunityComposition/all_pcoas_treat.pdf", width=9, height=4, units=c("in"), useDingbats=FALSE)

```


## Summary

- No differences in results between relative abundance & rarefied plots/stats
- Pairwise adonis significant between control treatment and all other variability treatments
- No lineage effects to suggest cryptic lineages host different microbiomes
- Beta dispersion not significantly different 


# ANCOM

Tutorial [here](https://github.com/FrederickHuangLin/ANCOM)

## Setup

```{r ancom packages}
library(readr)
library(tidyverse)
#library(dplyr)
library(nlme)
#install.packages('compositions')
library(compositions)
source("ancom_v2.1.R")
```
```{r ancom treatment lineages together, eval=FALSE}
otu_data_unt <- data.frame(ps.trim.rare@otu_table) 
otu_data <- data.frame(t(otu_data_unt))

meta_data = data.frame(ps.trim.rare@sam_data)
meta_data = meta_data %>% dplyr::rename(Sample.ID = frag)

# Step 1: Data pre-processing

feature_table = otu_data; sample_var = "Sample.ID"; group_var = NULL
out_cut = 0.05; zero_cut = 0.90; lib_cut = 900; neg_lb = FALSE
prepro = feature_table_pre_process(feature_table, meta_data, sample_var, group_var,out_cut, zero_cut, lib_cut, neg_lb)
feature_table = prepro$feature_table # Preprocessed feature table
meta_data = prepro$meta_data # Preprocessed metadata
struc_zero = prepro$structure_zeros # Structural zero info

# Step 2: ANCOM

main_var = "treat"; p_adj_method = "BH"; alpha = 0.05
adj_formula = NULL; rand_formula = NULL

res.all = ANCOM(feature_table, meta_data, struc_zero, main_var, p_adj_method, 
            alpha, adj_formula, rand_formula)

saveRDS(res.all,file="/Users/hannahaichelman/Documents/BU/TVE/16S_ITS2/16S_PreStress/CommunityComposition/ancom.res.treat.RDS")

# Step 3: Volcano Plot

# Number of taxa except structural zeros
n_taxa = ifelse(is.null(struc_zero), nrow(feature_table), sum(apply(struc_zero, 1, sum) == 0))
# Cutoff values for declaring differentially abundant taxa
cut_off = c(0.9 * (n_taxa -1), 0.8 * (n_taxa -1), 0.7 * (n_taxa -1), 0.6 * (n_taxa -1))
names(cut_off) = c("detected_0.9", "detected_0.8", "detected_0.7", "detected_0.6")

# Annotation data
dat_ann = data.frame(x = min(res.all$fig$data$x), y = cut_off["detected_0.6"], label = "W[0.6]")

fig.treat = res.all$fig +  
  geom_hline(yintercept = cut_off["detected_0.6"], linetype = "dashed") + 
  geom_text(data = dat_ann, aes(x = x, y = y, label = label), 
            size = 4, vjust = -0.5, hjust = 0, color = "orange", parse = TRUE)
fig.treat
```

```{r ancom lineage lineages together, eval=FALSE}
otu_data_unt <- data.frame(ps.trim.rare@otu_table)
otu_data <- data.frame(t(otu_data_unt))

meta_data = data.frame(ps.trim.rare@sam_data)
meta_data = meta_data %>% dplyr::rename(Sample.ID = frag)

# Step 1: Data pre-processing

feature_table = otu_data; sample_var = "Sample.ID"; group_var = NULL
out_cut = 0.05; zero_cut = 0.90; lib_cut = 900; neg_lb = FALSE
prepro = feature_table_pre_process(feature_table, meta_data, sample_var, group_var, 
                                   out_cut, zero_cut, lib_cut, neg_lb)
feature_table = prepro$feature_table # Preprocessed feature table
meta_data = prepro$meta_data # Preprocessed metadata
struc_zero = prepro$structure_zeros # Structural zero info

# Step 2: ANCOM

main_var = "lineage"; p_adj_method = "BH"; alpha = 0.05
adj_formula = NULL; rand_formula = NULL

res.all = ANCOM(feature_table, meta_data, struc_zero, main_var, p_adj_method, 
            alpha, adj_formula, rand_formula)

saveRDS(res.all,file="/Users/hannahaichelman/Documents/BU/TVE/16S_ITS2/16S_PreStress/CommunityComposition/ancom.res.lineage.RDS")

# Step 3: Volcano Plot

# Number of taxa except structural zeros
n_taxa = ifelse(is.null(struc_zero), nrow(feature_table), sum(apply(struc_zero, 1, sum) == 0))
# Cutoff values for declaring differentially abundant taxa
cut_off = c(0.9 * (n_taxa -1), 0.8 * (n_taxa -1), 0.7 * (n_taxa -1), 0.6 * (n_taxa -1))
names(cut_off) = c("detected_0.9", "detected_0.8", "detected_0.7", "detected_0.6")

# Annotation data
dat_ann = data.frame(x = min(res.all$fig$data$x), y = cut_off["detected_0.6"], label = "W[0.6]")

fig.lineage = res.all$fig +  
  geom_hline(yintercept = cut_off["detected_0.6"], linetype = "dashed") + 
  geom_text(data = dat_ann, aes(x = x, y = y, label = label), 
            size = 4, vjust = -0.5, hjust = 0, color = "orange", parse = TRUE)
fig.lineage
# two differentially abundant taxa

```

```{r ancom treatment lineages separate, eval=FALSE}
# separate by lineage - skipping this for now since there are no lineage differences for all asv's analyzed together. 

ps.trim.rare.L1 <- subset_samples(ps.trim.rare, lineage=="L1")
ps.trim.rare.L2 <- subset_samples(ps.trim,rare, lineage=="L2")

# Lineage 1 #
otu_data_unt.L1 <- data.frame(ps.trim.rare.L1@otu_table)
otu_data.L1 <- data.frame(t(otu_data_unt.L1))

meta_data.L1 = data.frame(ps.trim.rare.L1@sam_data)
meta_data.L1 = meta_data.L1 %>% rename(Sample.ID = frag)

# Step 1: Data pre-processing

feature_table.L1 = otu_data.L1; sample_var = "Sample.ID"; group_var = NULL
out_cut = 0.05; zero_cut = 0.90; lib_cut = 1000; neg_lb = FALSE
prepro.L1 = feature_table_pre_process(feature_table.L1, meta_data.L1, sample_var, group_var, 
                                   out_cut, zero_cut, lib_cut, neg_lb)
feature_table = prepro.L1$feature_table # Preprocessed feature table
meta_data = prepro.L1$meta_data # Preprocessed metadata
struc_zero = prepro.L1$structure_zeros # Structural zero info

# Step 2: ANCOM

main_var = "treat"; p_adj_method = "BH"; alpha = 0.05
adj_formula = NULL; rand_formula = NULL

res.all.L1 = ANCOM(feature_table, meta_data, struc_zero, main_var, p_adj_method, 
            alpha, adj_formula, rand_formula)

saveRDS(res.all.L1,file="/Users/hannahaichelman/Documents/BU/TVE/16S_ITS2/16S_PreStress/CommunityComposition/ancom.res.treat.L1.RDS")

# Step 3: Volcano Plot

# Number of taxa except structural zeros
n_taxa = ifelse(is.null(struc_zero), nrow(feature_table), sum(apply(struc_zero, 1, sum) == 0))
# Cutoff values for declaring differentially abundant taxa
cut_off = c(0.9 * (n_taxa -1), 0.8 * (n_taxa -1), 0.7 * (n_taxa -1), 0.6 * (n_taxa -1))
names(cut_off) = c("detected_0.9", "detected_0.8", "detected_0.7", "detected_0.6")

# Annotation data
dat_ann.L1 = data.frame(x = min(res.all.L1$fig$data$x), y = cut_off["detected_0.6"], label = "W[0.6]")

fig.treat.L1 = res.all.L1$fig +  
  geom_hline(yintercept = cut_off["detected_0.6"], linetype = "dashed") + 
  geom_text(data = dat_ann.L1, aes(x = x, y = y, label = label), 
            size = 4, vjust = -0.5, hjust = 0, color = "orange", parse = TRUE)
fig.treat.L1 


# Lineage 2 #
otu_data_unt.L2 <- data.frame(ps.trim.L2@otu_table)
otu_data.L2 <- data.frame(t(otu_data_unt.L2))

meta_data.L2 = data.frame(ps.trim.L2@sam_data)
meta_data.L2 = meta_data.L2 %>% rename(Sample.ID = frag)

# Step 1: Data pre-processing

feature_table.L2 = otu_data.L2; sample_var = "Sample.ID"; group_var = NULL
out_cut = 0.05; zero_cut = 0.90; lib_cut = 1000; neg_lb = FALSE
prepro.L2 = feature_table_pre_process(feature_table.L2, meta_data.L2, sample_var, group_var, 
                                   out_cut, zero_cut, lib_cut, neg_lb)
feature_table = prepro.L2$feature_table # Preprocessed feature table
meta_data = prepro.L2$meta_data # Preprocessed metadata
struc_zero = prepro.L2$structure_zeros # Structural zero info

# Step 2: ANCOM

main_var = "treat"; p_adj_method = "BH"; alpha = 0.05
adj_formula = NULL; rand_formula = NULL

res.all.L2 = ANCOM(feature_table, meta_data, struc_zero, main_var, p_adj_method, 
            alpha, adj_formula, rand_formula)

saveRDS(res.all.L2,file="/Users/hannahaichelman/Documents/BU/TVE/16S_ITS2/16S_PreStress/CommunityComposition/ancom.res.treat.L2.RDS")

# Step 3: Volcano Plot

# Number of taxa except structural zeros
n_taxa = ifelse(is.null(struc_zero), nrow(feature_table), sum(apply(struc_zero, 1, sum) == 0))
# Cutoff values for declaring differentially abundant taxa
cut_off = c(0.9 * (n_taxa -1), 0.8 * (n_taxa -1), 0.7 * (n_taxa -1), 0.6 * (n_taxa -1))
names(cut_off) = c("detected_0.9", "detected_0.8", "detected_0.7", "detected_0.6")

# Annotation data
dat_ann.L2 = data.frame(x = min(res.all.L2$fig$data$x), y = cut_off["detected_0.6"], label = "W[0.6]")

fig.treat.L2 = res.all.L2$fig +  
  geom_hline(yintercept = cut_off["detected_0.6"], linetype = "dashed") + 
  geom_text(data = dat_ann.L2, aes(x = x, y = y, label = label), 
            size = 4, vjust = -0.5, hjust = 0, color = "orange", parse = TRUE)
fig.treat.L2 
```


```{r ancom sitename lineages separate, eval=FALSE}
# Lineage 1 #
otu_data_unt.L1 <- data.frame(ps.trim.L1@otu_table)
otu_data.L1 <- data.frame(t(otu_data_unt.L1))

meta_data.L1 = data.frame(ps.trim.L1@sam_data)
meta_data.L1 = meta_data.L1 %>% rename(Sample.ID = frag)

# Step 1: Data pre-processing

feature_table.L1 = otu_data.L1; sample_var = "Sample.ID"; group_var = NULL
out_cut = 0.05; zero_cut = 0.90; lib_cut = 1000; neg_lb = FALSE
prepro.L1 = feature_table_pre_process(feature_table.L1, meta_data.L1, sample_var, group_var, 
                                   out_cut, zero_cut, lib_cut, neg_lb)
feature_table = prepro.L1$feature_table # Preprocessed feature table
meta_data = prepro.L1$meta_data # Preprocessed metadata
struc_zero = prepro.L1$structure_zeros # Structural zero info

# Step 2: ANCOM

main_var = "sitename"; p_adj_method = "BH"; alpha = 0.05
adj_formula = NULL; rand_formula = NULL

res.all.L1 = ANCOM(feature_table, meta_data, struc_zero, main_var, p_adj_method, 
            alpha, adj_formula, rand_formula)

saveRDS(res.all.L1,file="/Users/hannahaichelman/Documents/BU/TVE/16S_ITS2/16S_PreStress/CommunityComposition/ancom.res.site.L1.RDS")

# Step 3: Volcano Plot

# Number of taxa except structural zeros
n_taxa = ifelse(is.null(struc_zero), nrow(feature_table), sum(apply(struc_zero, 1, sum) == 0))
# Cutoff values for declaring differentially abundant taxa
cut_off = c(0.9 * (n_taxa -1), 0.8 * (n_taxa -1), 0.7 * (n_taxa -1), 0.6 * (n_taxa -1))
names(cut_off) = c("detected_0.9", "detected_0.8", "detected_0.7", "detected_0.6")

# Annotation data
dat_ann.L1 = data.frame(x = min(res.all.L1$fig$data$x), y = cut_off["detected_0.6"], label = "W[0.6]")

fig.treat.L1 = res.all.L1$fig +  
  geom_hline(yintercept = cut_off["detected_0.6"], linetype = "dashed") + 
  geom_text(data = dat_ann.L1, aes(x = x, y = y, label = label), 
            size = 4, vjust = -0.5, hjust = 0, color = "orange", parse = TRUE)
fig.treat.L1 
# no differentially abundant taxa

# Lineage 2 #
otu_data_unt.L2 <- data.frame(ps.trim.L2@otu_table)
otu_data.L2 <- data.frame(t(otu_data_unt.L2))

meta_data.L2 = data.frame(ps.trim.L2@sam_data)
meta_data.L2 = meta_data.L2 %>% rename(Sample.ID = frag)

# Step 1: Data pre-processing

feature_table.L2 = otu_data.L2; sample_var = "Sample.ID"; group_var = NULL
out_cut = 0.05; zero_cut = 0.90; lib_cut = 1000; neg_lb = FALSE
prepro.L2 = feature_table_pre_process(feature_table.L2, meta_data.L2, sample_var, group_var, 
                                   out_cut, zero_cut, lib_cut, neg_lb)
feature_table = prepro.L2$feature_table # Preprocessed feature table
meta_data = prepro.L2$meta_data # Preprocessed metadata
struc_zero = prepro.L2$structure_zeros # Structural zero info

# Step 2: ANCOM

main_var = "sitename"; p_adj_method = "BH"; alpha = 0.05
adj_formula = NULL; rand_formula = NULL

res.all.L2 = ANCOM(feature_table, meta_data, struc_zero, main_var, p_adj_method, 
            alpha, adj_formula, rand_formula)

saveRDS(res.all.L2,file="/Users/hannahaichelman/Documents/BU/TVE/16S_ITS2/16S_PreStress/CommunityComposition/ancom.res.site.L2.RDS")

# Step 3: Volcano Plot

# Number of taxa except structural zeros
n_taxa = ifelse(is.null(struc_zero), nrow(feature_table), sum(apply(struc_zero, 1, sum) == 0))
# Cutoff values for declaring differentially abundant taxa
cut_off = c(0.9 * (n_taxa -1), 0.8 * (n_taxa -1), 0.7 * (n_taxa -1), 0.6 * (n_taxa -1))
names(cut_off) = c("detected_0.9", "detected_0.8", "detected_0.7", "detected_0.6")

# Annotation data
dat_ann.L2 = data.frame(x = min(res.all.L2$fig$data$x), y = cut_off["detected_0.6"], label = "W[0.6]")

fig.treat.L2 = res.all.L2$fig +  
  geom_hline(yintercept = cut_off["detected_0.6"], linetype = "dashed") + 
  geom_text(data = dat_ann.L2, aes(x = x, y = y, label = label), 
            size = 4, vjust = -0.5, hjust = 0, color = "orange", parse = TRUE)
fig.treat.L2
# No differentially abundant taxa across site of origin for either lineage
```


## Synthesizing ANCOM results

Re-read in data

```{r load ancom results}
res.treat <- readRDS("/Users/hannahaichelman/Documents/BU/TVE/16S_ITS2/16S_PreStress/CommunityComposition/ancom.res.treat.RDS")

res.lineage <- readRDS("/Users/hannahaichelman/Documents/BU/TVE/16S_ITS2/16S_PreStress/CommunityComposition/ancom.res.lineage.RDS")

```

Which ones are 'significant'

```{r subset out files}
res.out.treat <- res.treat$out
res.out.treat.sig <- res.out.treat[res.out.treat$detected_0.6==TRUE,]
#4 seqs (sq13, sq33, sq38, sq81)

res.out.lineage <- res.lineage$out
res.out.lineage.sig <- res.out.lineage[res.out.lineage$detected_0.6==TRUE,]
#2 seqs (sq4, sq19)

```

Subset the sig ones in phyloseq

```{r}
want.treat <- c(res.out.treat.sig$taxa_id)

want.lineage <- c(res.out.lineage.sig$taxa_id)

ps.sig.taxa.treat <- subset_taxa(ps.trim.rare,row.names(ps.trim.rare@tax_table) %in% want.treat)
# Taxonomy Table:     [4 taxa by 7 taxonomic ranks]:
#      Kingdom    Phylum            Class                 Order                 Family               
# sq13 "Bacteria" "Proteobacteria"  "Alphaproteobacteria" "SAR11 clade"         "Clade I"            
# sq33 "Bacteria" "Proteobacteria"  "Alphaproteobacteria" "SAR11 clade"         "Clade III"          
# sq38 "Bacteria" "Bacteroidota"    "Bacteroidia"         "Flavobacteriales"    "Flavobacteriaceae"  
# sq81 "Bacteria" "Patescibacteria" "Parcubacteria"       "Class_Parcubacteria" "Class_Parcubacteria"
#      Genus                 Species              
# sq13 "Clade Ia"            "Genus_Clade Ia"     
# sq33 "Family_Clade III"    "Family_Clade III"   
# sq38 "Tenacibaculum"       "mesophilum"         
# sq81 "Class_Parcubacteria" "Class_Parcubacteria"

ps.sig.taxa.lineage <- subset_taxa(ps.trim.rare,row.names(ps.trim.rare@tax_table) %in% want.lineage)
# Taxonomy Table:     [2 taxa by 7 taxonomic ranks]:
#      Kingdom    Phylum           Class                 Order             Family               Genus             
# sq5  "Bacteria" "Proteobacteria" "Gammaproteobacteria" "Xanthomonadales" "Rhodanobacteraceae" "Dyella"          
# sq33 "Bacteria" "Proteobacteria" "Alphaproteobacteria" "SAR11 clade"     "Clade III"          "Family_Clade III"
#      Species           
# sq5  "thiooxydans"     
# sq33 "Family_Clade III"

#for variability treatments
plot = plot_bar(ps.sig.taxa.treat,x="treat",y="Abundance",fill="Order")+
  facet_wrap(~Family,scales="free")
ggsave(plot, file = "/Users/hannahaichelman/Documents/BU/TVE/16S_ITS2/16S_PreStress/CommunityComposition/sig.family.abun.treat.pdf",width=8)

# Clade I and Clade III are SAR11, the most abundant plankton in the ocean. Evidence that P. astreoides eats them in mesocosm experiments: https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0229442
# Tenacibaculum in corals: https://journals.asm.org/doi/full/10.1128/mSystems.01086-20
# Alteromonas found being released by spawning colonies supposedly to provision offspring: https://link.springer.com/article/10.1007/s00248-012-0105-z

#for lineage
plot = plot_bar(ps.sig.taxa.lineage,x="lineage",y="Abundance",fill="Family")+
  facet_wrap(~Family,scales="free")
ggsave(plot, file = "/Users/hannahaichelman/Documents/BU/TVE/16S_ITS2/16S_PreStress/CommunityComposition/sig.family.abun.lineage.pdf",width=8)

# Rhodanobacteraceae (Dyella thiooxydans) - thiosulfate (oxidizing, found in sunflower fields in Korea

```


## Make heat map of ANCOM results

```{r}
#https://schuyler-smith.github.io/phylosmith/graphs.html

#devtools::install_github('schuyler-smith/phylosmith')
library(phylosmith)

# treatment heat map
ps.treat.glom <- tax_glom(ps.sig.taxa.treat, "Family")
ps1 <- merge_samples(ps.treat.glom, "treat")

heatmap = abundance_heatmap(ps1, treatment = c('treat'), 
                  classification = "Family", transformation = "log",
                  treatment_labels = c("Control","Low Var","Mod Var","High Var"),
                  sample_labels = NULL)
#                  classification_labels = c("SARII Clade I", "Sar11 Clade III",
#                                            "Order_Flavobacteriales", "Class_Parcubacteria"))
heatmap
ggsave(heatmap, file="/Users/hannahaichelman/Documents/BU/TVE/16S_ITS2/16S_PreStress/CommunityComposition/heatmap.treat.pdf",width=6, height=4, units=c("in"), useDingbats=FALSE)

# lineage heat map
ps.lin.glom <- tax_glom(ps.sig.taxa.lineage, "Family")
ps1 <- merge_samples(ps.lin.glom, "lineage")

heatmap = abundance_heatmap(ps1, treatment = c('lineage'), 
                  classification = "Family", transformation = "log",
                  treatment_labels = c("L1","L2"),
                  sample_labels = NULL)
#                  classification_labels = c("SARII Clade I", "Sar11 Clade III",
#                                            "Order_Flavobacteriales", "Class_Parcubacteria"))
heatmap
ggsave(heatmap, file="/Users/hannahaichelman/Documents/BU/TVE/16S_ITS2/16S_PreStress/CommunityComposition/heatmap.lineage.pdf",width=6, height=4, units=c("in"), useDingbats=FALSE)

# combine with pcoa's to make big figure
pcoas.combined = ggarrange(pcoa.treat.rel, plot.core.rel.treat, pcoa.acc.all.rel.treat,
          labels = c("(a) All ASVs", "(b) Core ASVs", "(c) Accessory ASVs"),
          common.legend = TRUE, legend = "right",
          ncol = 2, nrow = 2)

ggsave(pcoas.combined, filename = "/Users/hannahaichelman/Documents/BU/TVE/16S_ITS2/16S_PreStress/CommunityComposition/all_pcoas_treat.pdf", width=8, height=6, units=c("in"), useDingbats=FALSE)

# make final lineage pca's

pcoas.combined.lin = ggarrange(pcoa.lineage.rel, plot.core.rel.lin, pcoa.acc.all.rel.lin,
          labels = c("(a) All ASVs", "(b) Core ASVs", "(c) Accessory ASVs"),
          common.legend = TRUE, legend = "right",
          ncol = 2, nrow = 2)

ggsave(pcoas.combined.lin, filename = "/Users/hannahaichelman/Documents/BU/TVE/16S_ITS2/16S_PreStress/CommunityComposition/all_pcoas_lineage.pdf", width=8, height=6, units=c("in"), useDingbats=FALSE)

```

