---
title: "TVE 16S Diversity Analyses"
author: "Nicola Kriefall, updated by Hannah Aichelman"
date: "3/12/2024"
output:
 rmarkdown::html_document:
    theme: cerulean
    toc: yes
    toc_float: yes
    highlight: haddock
    number_sections: true
---

```{r setup, include=FALSE}
# set to github repository location
setwd("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability")
```

# Setup

## Packages

```{r packages}
library(ggplot2)
library(cowplot)
library(phyloseq)
library(car)
library(ggpubr)
library(vegan)
library(dada2)
library(tidyverse)
```


## Read in Phyloseq objects

```{r making phyloseq objects, eval=FALSE}
# can skip this section once you've run it once and saved output Rdata files
load("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/taxa2.Rdata")

#cleaned, rarefied files:
ps.rare.1k.t0 = readRDS("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/phyloseq.t0.rarefied.1k.rds")
seqtab.rare.1k.t0 <- data.frame(ps.rare.1k.t0@otu_table)
samdf.rare.1k.t0 <- data.frame(ps.rare.1k.t0@sam_data)

ps.rare.1k.prestress = readRDS("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/phyloseq.prestress.rarefied.1k.rds")
seqtab.rare.1k.prestress <- data.frame(ps.rare.1k.prestress@otu_table)
samdf.rare.1k.prestress <- data.frame(ps.rare.1k.prestress@sam_data)

ps.rare.t0 <- phyloseq(otu_table(seqtab.rare.1k.t0, taxa_are_rows=FALSE), 
                    sample_data(samdf.rare.1k.t0), 
                    tax_table(taxa2))
ps.rare.t0 

ps.rare.prestress <- phyloseq(otu_table(seqtab.rare.1k.prestress, taxa_are_rows=FALSE), 
                    sample_data(samdf.rare.1k.prestress), 
                    tax_table(taxa2))
ps.rare.prestress 

save(ps.rare.t0,file="ps.rare.t0.Rdata")
save(ps.rare.prestress,file="ps.rare.prestress.Rdata")

#cleaned, unrarefied files:
ps.clean.t0 = readRDS("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/phyloseq.cleanest.t0.rds")
seqtab.clean.t0 <- data.frame(ps.clean.t0@otu_table)
samdf.clean.t0 <- data.frame(ps.clean.t0@sam_data)

ps.clean.prestress = readRDS("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/phyloseq.cleanest.prestress.rds")
seqtab.clean.prestress <- data.frame(ps.clean.prestress@otu_table)
samdf.clean.prestress <- data.frame(ps.clean.prestress@sam_data)

ps.clean.t0 <- phyloseq(otu_table(seqtab.clean.t0, taxa_are_rows=FALSE), 
                    sample_data(samdf.clean.t0), 
                    tax_table(taxa2))
ps.clean.t0

ps.clean.prestress <- phyloseq(otu_table(seqtab.clean.prestress, taxa_are_rows=FALSE), 
                    sample_data(samdf.clean.prestress), 
                    tax_table(taxa2))
ps.clean.prestress

save(ps.clean.t0,file="ps.clean.t0.Rdata")
save(ps.clean.prestress,file="ps.clean.prestress.Rdata")


##TRIMMED VERSIONS:
# trimmed and rarefied
ps.trim.rare.t0 = readRDS("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/phyloseq.t0.trim.rarefied.1k.rds")
seqtab.trim.rare.t0 <- data.frame(ps.trim.rare.t0@otu_table)
samdf.trim.rare.t0 <- data.frame(ps.trim.rare.t0@sam_data)

ps.trim.rare.prestress = readRDS("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/phyloseq.prestress.trim.rarefied.1k.rds")
seqtab.trim.rare.prestress <- data.frame(ps.trim.rare.prestress@otu_table)
samdf.trim.rare.prestress <- data.frame(ps.trim.rare.prestress@sam_data)

ps.trim.rare.t0 <- phyloseq(otu_table(seqtab.trim.rare.t0, taxa_are_rows=FALSE), 
                    sample_data(samdf.trim.rare.t0), 
                    tax_table(taxa2))
ps.trim.rare.t0

ps.trim.rare.prestress <- phyloseq(otu_table(seqtab.trim.rare.prestress, taxa_are_rows=FALSE), 
                    sample_data(samdf.trim.rare.prestress), 
                    tax_table(taxa2))
ps.trim.rare.prestress

save(ps.trim.rare.t0,file="ps.trim.rare.t0.Rdata")
save(ps.trim.rare.prestress,file="ps.trim.rare.prestress.Rdata")

# just trimmed
ps.trim.t0 = readRDS("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/phyloseq.cleanest.trim.t0.rds")
seqtab.trim.t0 <- data.frame(ps.trim.t0@otu_table)
samdf.trim.t0 <- data.frame(ps.trim.t0@sam_data)

ps.trim.prestress = readRDS("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/phyloseq.cleanest.trim.prestress.rds")
seqtab.trim.prestress <- data.frame(ps.trim.prestress@otu_table)
samdf.trim.prestress <- data.frame(ps.trim.prestress@sam_data)

ps.trim.t0 <- phyloseq(otu_table(seqtab.trim.t0, taxa_are_rows=FALSE), 
                    sample_data(samdf.trim.t0), 
                    tax_table(taxa2))
ps.trim.t0

ps.trim.prestress <- phyloseq(otu_table(seqtab.trim.prestress, taxa_are_rows=FALSE), 
                    sample_data(samdf.trim.prestress), 
                    tax_table(taxa2))
ps.trim.prestress

save(ps.trim.t0,file="ps.trim.t0.Rdata")
save(ps.trim.prestress,file="ps.trim.prestress.Rdata")
```

```{r read in ps objects}
load("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/taxa2.Rdata")
# can skip to here once you have made the .Rdata files above
load("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/ps.clean.t0.Rdata")
load("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/ps.clean.prestress.Rdata")

load("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/ps.rare.t0.Rdata")
load("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/ps.rare.prestress.Rdata")

load("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/ps.trim.rare.t0.Rdata")
load("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/ps.trim.rare.prestress.Rdata")

load("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/ps.trim.t0.Rdata")
load("~/Dropbox/BU/TVE/TVE_Github/DielTempVariability/16S_Microbiome/data_files/ps.trim.prestress.Rdata")


# remove lineage NA's from T0 data
ps.clean.t0 <- subset_samples(ps.clean.t0,(!is.na(lineage)))

ps.rare.t0 <- subset_samples(ps.rare.t0,(!is.na(lineage)))

ps.trim.rare.t0 <- subset_samples(ps.trim.rare.t0,(!is.na(lineage)))

ps.trim.t0 <- subset_samples(ps.trim.t0,(!is.na(lineage)))


# remove Control 2, low variability, and high variability treatments, and lineage NA's from prestress data
ps.clean.prestress <- subset_samples(ps.clean.prestress,(!is.na(lineage)))
ps.clean.prestress <- subset_samples(ps.clean.prestress,(!is.na(treat)))
ps.clean.prestress <- subset_samples(ps.clean.prestress,(treat!="Control 2"))
ps.clean.prestress <- subset_samples(ps.clean.prestress,(treat!="Low Var"))
ps.clean.prestress <- subset_samples(ps.clean.prestress,(treat!="High Var"))

ps.rare.prestress <- subset_samples(ps.rare.prestress,(!is.na(lineage)))
ps.rare.prestress <- subset_samples(ps.rare.prestress,(!is.na(treat)))
ps.rare.prestress <- subset_samples(ps.rare.prestress,(treat!="Control 2"))
ps.rare.prestress <- subset_samples(ps.rare.prestress,(treat!="Low Var"))
ps.rare.prestress <- subset_samples(ps.rare.prestress,(treat!="High Var"))

ps.trim.rare.prestress <- subset_samples(ps.trim.rare.prestress,(!is.na(lineage)))
ps.trim.rare.prestress <- subset_samples(ps.trim.rare.prestress,(!is.na(treat)))
ps.trim.rare.prestress <- subset_samples(ps.trim.rare.prestress,(treat!="Control 2"))
ps.trim.rare.prestress <- subset_samples(ps.trim.rare.prestress,(treat!="Low Var"))
ps.trim.rare.prestress <- subset_samples(ps.trim.rare.prestress,(treat!="High Var"))

ps.trim.prestress <- subset_samples(ps.trim.prestress,(!is.na(lineage)))
ps.trim.prestress <- subset_samples(ps.trim.prestress,(!is.na(treat)))
ps.trim.prestress <- subset_samples(ps.trim.prestress,(treat!="Control 2"))
ps.trim.prestress <- subset_samples(ps.trim.prestress,(treat!="Low Var"))
ps.trim.prestress <- subset_samples(ps.trim.prestress,(treat!="High Var"))
```

# Diversity

[Notes from phyloseq author](https://rdrr.io/bioc/phyloseq/man/estimate_richness.html)
Visualize alpha-diversity - Should be done on raw, untrimmed dataset

```{r generate diversity metrics T0}
df.t0 <- data.frame(estimate_richness(ps.clean.t0, split=TRUE, measures=c("Shannon","InvSimpson","Observed")))
samdf.clean.t0 <- data.frame(ps.clean.t0@sam_data)

df.t0$frag <- rownames(df.t0)
df.t0.div <- merge(df.t0,samdf.clean.t0,by="frag") #add sample data

#shannon diversity divided by species richness
df.t0.div$even <- df.t0.div$Shannon/(log(df.t0.div$Observed))

str(df.t0.div)
df.t0.div$lineage = as.factor(df.t0.div$lineage)
df.t0.div$sitename = as.factor(df.t0.div$sitename)
```

```{r generate diversity metrics Prestress}
df.ps <- data.frame(estimate_richness(ps.clean.prestress, split=TRUE, measures=c("Shannon","InvSimpson","Observed")))
samdf.clean.ps <- data.frame(ps.clean.prestress@sam_data)

df.ps$frag <- rownames(df.ps)
df.ps.div <- merge(df.ps,samdf.clean.ps,by="frag") #add sample data

#shannon diversity divided by species richness
df.ps.div$even <- df.ps.div$Shannon/(log(df.ps.div$Observed))

str(df.ps.div)
df.ps.div$treat = as.factor(df.ps.div$treat)
df.ps.div$lineage = as.factor(df.ps.div$lineage)
df.ps.div$sitename = as.factor(df.ps.div$sitename)
```
## Alpha metrics {.tabset}

Looking at everything individually, more of a synthesis below
Alpha diversity is within-sample diversity. Beta diversity, on the other hand, is a measure of similarity or dissimilarity between populations/groups. 

### Shannon
Shannon diversity estimates species diversity, considers number of species (richness) and their relative abundance (evenness)

```{r shannon lineage t0 and prestress}
#plotting lineages
cols_lineage <- c("L1" = "#3f007d", "L2" = "#807dba", "L3" = "#bcbddc")

gg.lin.sh.t0 <- ggplot(df.t0.div, aes(x=lineage, y=Shannon,color=lineage))+
  geom_boxplot(outlier.shape=NA)+
  scale_colour_manual(values=cols_lineage)+
  geom_jitter(alpha=0.5)+
  xlab("Lineage")+
  ylab("Shannon index")+
  theme_bw()+
  #guides(color=guide_legend(title="Reef zone"),shape=guide_legend(title="Reef zone"))+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none") 
gg.lin.sh.t0

gg.lin.sh.ps <- ggplot(df.ps.div, aes(x=lineage, y=Shannon,color=lineage))+
  geom_boxplot(outlier.shape=NA)+
  scale_colour_manual(values=cols_lineage)+
  geom_jitter(alpha=0.5)+
  xlab("Lineage")+
  ylab("Shannon index")+
  theme_bw()+
  #guides(color=guide_legend(title="Reef zone"),shape=guide_legend(title="Reef zone"))+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none") 
gg.lin.sh.ps

```


```{r shannon site t0 and prestress}
cols_site_diverging <- c("CI" = "#543005", "PD"= "#bf812d",  "SP"= "#dfc27d",  "BN" = "#003c30", "BS"= "#35978f", "CA"= "#80cdc1")

gg.site.sha.t0 <- ggplot(df.t0.div,aes(x=sitename,y=Shannon,color=sitename))+
  geom_boxplot(outlier.shape=NA)+
  scale_color_manual(values=cols_site_diverging)+
  geom_jitter(alpha=0.5)+
  ylab("Shannon index")+
  xlab("Site")+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none")
gg.site.sha.t0

gg.site.sha.ps <- ggplot(df.ps.div,aes(x=sitename,y=Shannon,color=sitename))+
  geom_boxplot(outlier.shape=NA)+
  scale_color_manual(values=cols_site_diverging)+
  geom_jitter(alpha=0.5)+
  ylab("Shannon index")+
  xlab("Site")+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none")
gg.site.sha.ps
```



```{r shannon treatment - prestress only}
#plotting treatment
cols_treat_reds <- c("darkgrey", "#CC3300")

gg.treat.sh.ps <- ggplot(df.ps.div, aes(x=treat, y=Shannon,color=treat))+
  geom_boxplot(outlier.shape=NA)+
  scale_colour_manual(values=cols_treat_reds)+
  geom_jitter(alpha=0.5)+
  xlab("Treatment")+
  ylab("Shannon index")+
  theme_bw()+
  #guides(color=guide_legend(title="Reef zone"),shape=guide_legend(title="Reef zone"))+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none")
gg.treat.sh.ps

#facet wrap by lineage
gg.treat.lin.sh.ps <- ggplot(df.ps.div, aes(x=treat, y=Shannon,color=treat))+
  geom_boxplot(outlier.shape=NA)+
  scale_colour_manual(values=cols_treat_reds)+
  geom_jitter(alpha=0.5)+
  xlab("Treatment")+
  ylab("Shannon index")+
  theme_bw()+
  #guides(color=guide_legend(title="Reef zone"),shape=guide_legend(title="Reef zone"))+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none")+
  facet_wrap(~lineage)
gg.treat.lin.sh.ps
```


```{r shannon stats}
#Shannon
shapiro.test(df.t0.div$Shannon) # not fine 
leveneTest(df.t0.div$Shannon~lineage,data=df.t0.div) #fine

a.div.t0 <- aov(Shannon~lineage,data=df.t0.div)
summary(a.div.t0) 
#             Df Sum Sq Mean Sq F value Pr(>F)
# lineage      2  2.114  1.0569   1.685  0.196
# Residuals   47 29.477  0.6272               


shapiro.test(df.ps.div$Shannon) # not fine 
leveneTest(df.ps.div$Shannon~lineage,data=df.ps.div) #fine

a.div.ps <- aov(Shannon~lineage+treat,data=df.ps.div)
summary(a.div.ps) 
#             Df Sum Sq Mean Sq F value Pr(>F)
# lineage      2   1.49  0.7454   0.642  0.529
# treat        1   1.29  1.2878   1.109  0.295
# Residuals   86  99.85  1.1611               
```

### Simpson

Simpson index considered more of a dominance index, accounts for proportion of species in a sample.
```{r simpson lineage - t0 and prestress}
gg.lin.sim.t0 <- ggplot(df.t0.div, aes(x=lineage, y=InvSimpson,color=lineage))+
  geom_boxplot(outlier.shape=NA)+
  scale_colour_manual(values=cols_lineage)+
  geom_jitter(alpha=0.5)+
  xlab("Lineage")+
  ylab("Inv. Simpson index")+
  theme_bw()+
  #guides(color=guide_legend(title="Reef zone"),shape=guide_legend(title="Reef zone"))+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none")
gg.lin.sim.t0

gg.lin.sim.ps <- ggplot(df.ps.div, aes(x=lineage, y=InvSimpson,color=lineage))+
  geom_boxplot(outlier.shape=NA)+
  scale_colour_manual(values=cols_lineage)+
  geom_jitter(alpha=0.5)+
  xlab("Lineage")+
  ylab("Inv. Simpson index")+
  theme_bw()+
  #guides(color=guide_legend(title="Reef zone"),shape=guide_legend(title="Reef zone"))+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none")
gg.lin.sim.ps
```

```{r simpson site - T0 and prestress}
gg.site.sim.t0 <- ggplot(df.t0.div,aes(x=sitename,y=InvSimpson,color=sitename))+
  geom_boxplot(outlier.shape=NA)+
  scale_color_manual(values=cols_site_diverging)+
  geom_jitter(alpha=0.5)+
  ylab("Inv. Simpson index")+
  xlab("Site")+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none") 
gg.site.sim.t0

gg.site.sim.ps <- ggplot(df.ps.div,aes(x=sitename,y=InvSimpson,color=sitename))+
  geom_boxplot(outlier.shape=NA)+
  scale_color_manual(values=cols_site_diverging)+
  geom_jitter(alpha=0.5)+
  ylab("Inv. Simpson index")+
  xlab("Site")+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none") 
gg.site.sim.ps
```



```{r simpson treatment - prestress only}
#plotting treatment
gg.treat.sim.ps <- ggplot(df.ps.div, aes(x=treat, y=InvSimpson,color=treat))+
  geom_boxplot(outlier.shape=NA)+
  scale_colour_manual(values=cols_treat_reds)+
  geom_jitter(alpha=0.5)+
  xlab("Treatment")+
  ylab("Inv. Simpson index")+
  theme_bw()+
  #guides(color=guide_legend(title="Reef zone"),shape=guide_legend(title="Reef zone"))+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none") 
gg.treat.sim.ps

gg.treat.lin.sim.ps <- ggplot(df.ps.div, aes(x=treat, y=InvSimpson,color=treat))+
  geom_boxplot(outlier.shape=NA)+
  scale_colour_manual(values=cols_treat_reds)+
  geom_jitter(alpha=0.5)+
  xlab("Treatment")+
  ylab("Inv. Simpson index")+
  theme_bw()+
  #guides(color=guide_legend(title="Reef zone"),shape=guide_legend(title="Reef zone"))+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none")+
  facet_wrap(~lineage)
gg.treat.lin.sim.ps
```

```{r simpson stats}
#simpson
shapiro.test(df.t0.div$InvSimpson) #fine
leveneTest(df.t0.div$InvSimpson~lineage,data=df.t0.div) #fine 

a.div <- aov(InvSimpson~lineage,data=df.t0.div)
summary(a.div) 
#             Df Sum Sq Mean Sq F value Pr(>F)  
# lineage      2   8904    4452   2.518 0.0914 .
# Residuals   47  83085    1768                 

#simpson
shapiro.test(df.ps.div$InvSimpson) #not fine
leveneTest(df.ps.div$InvSimpson~lineage,data=df.ps.div) #fine 

a.div <- aov(InvSimpson~lineage+treat,data=df.ps.div)
summary(a.div) 
#             Df Sum Sq Mean Sq F value Pr(>F)
# lineage      2    117   58.74   0.194  0.824
# treat        1    164  163.50   0.541  0.464
# Residuals   86  25973  302.01               
```

### Richness

```{r richness lineage - T0 and prestress}
gg.lin.obs.t0 <- ggplot(df.t0.div, aes(x=lineage, y=Observed,color=lineage))+
  geom_boxplot(outlier.shape=NA)+
  scale_colour_manual(values=cols_lineage)+
  geom_jitter(alpha=0.5)+
  xlab("Lineage")+
  ylab("ASV richness")+
  theme_bw()+
  #guides(color=guide_legend(title="Reef zone"),shape=guide_legend(title="Reef zone"))+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none") 
gg.lin.obs.t0

gg.lin.obs.ps <- ggplot(df.ps.div, aes(x=lineage, y=Observed,color=lineage))+
  geom_boxplot(outlier.shape=NA)+
  scale_colour_manual(values=cols_lineage)+
  geom_jitter(alpha=0.5)+
  xlab("Lineage")+
  ylab("ASV richness")+
  theme_bw()+
  #guides(color=guide_legend(title="Reef zone"),shape=guide_legend(title="Reef zone"))+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none") 
gg.lin.obs.ps

```

```{r richness site - T0 and prestress}
gg.site.obs.t0 <- ggplot(df.t0.div,aes(x=sitename,y=Observed,color=sitename))+
  geom_boxplot(outlier.shape=NA)+
  scale_color_manual(values=cols_site_diverging)+
  geom_jitter(alpha=0.5)+
  ylab("ASV richness")+
  xlab("Site")+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none") 
gg.site.obs.t0

gg.site.obs.ps <- ggplot(df.ps.div,aes(x=sitename,y=Observed,color=sitename))+
  geom_boxplot(outlier.shape=NA)+
  scale_color_manual(values=cols_site_diverging)+
  geom_jitter(alpha=0.5)+
  ylab("ASV richness")+
  xlab("Site")+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none") 
gg.site.obs.ps
```

```{r richness treatment - prestress only}
#plotting treatment
gg.treat.obs.ps <- ggplot(df.ps.div, aes(x=treat, y=Observed,color=treat))+
  geom_boxplot(outlier.shape=NA)+
  scale_colour_manual(values=cols_treat_reds)+
  geom_jitter(alpha=0.5)+
  xlab("Treatment")+
  ylab("ASV richness")+
  theme_bw()+
  #guides(color=guide_legend(title="Reef zone"),shape=guide_legend(title="Reef zone"))+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none") 
gg.treat.obs.ps

gg.treat.lin.obs <- ggplot(df.ps.div, aes(x=treat, y=Observed,color=treat))+
  geom_boxplot(outlier.shape=NA)+
  scale_colour_manual(values=cols_treat_reds)+
  geom_jitter(alpha=0.5)+
  xlab("Treatment")+
  ylab("ASV Richness")+
  theme_bw()+
  #guides(color=guide_legend(title="Reef zone"),shape=guide_legend(title="Reef zone"))+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none")+
  facet_wrap(~lineage)
gg.treat.lin.obs
```

```{r richness stats}
#otu richness
shapiro.test(df.t0.div$Observed) #nope
df.t0.div$obs.log <- log(df.t0.div$Observed)
shapiro.test(df.t0.div$obs.log) #still no...
leveneTest(df.t0.div$obs.log~lineage,data=df.t0.div) #fine

a.div <- aov(Observed~lineage,data=df.t0.div)
summary(a.div) 
#             Df  Sum Sq Mean Sq F value Pr(>F)
# lineage      2    8431    4216   0.185  0.831
# Residuals   47 1069504   22755               

a.div <- aov(Observed~lineage+treat,data=df.ps.div)
summary(a.div) 
#             Df  Sum Sq Mean Sq F value Pr(>F)  
# lineage      2    4255    2128   0.180 0.8353  
# treat        1   75711   75711   6.417 0.0131 *
# Residuals   86 1014711   11799                 
```

### Evenness

```{r evenness site - T0 and prestress}
gg.site.eve.t0 <- ggplot(df.t0.div,aes(x=sitename,y=even,color=sitename))+
  geom_boxplot(outlier.shape=NA)+
  scale_color_manual(values=cols_site_diverging)+
  geom_jitter(alpha=0.5)+
  ylab("Evenness")+
  xlab("Site")+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none") 
gg.site.eve.t0

gg.site.eve.ps <- ggplot(df.ps.div,aes(x=sitename,y=even,color=sitename))+
  geom_boxplot(outlier.shape=NA)+
  scale_color_manual(values=cols_site_diverging)+
  geom_jitter(alpha=0.5)+
  ylab("Evenness")+
  xlab("Site")+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none") 
gg.site.eve.ps
```

```{r evenness lineage - T0 and prestress}
gg.lin.eve.t0 <- ggplot(df.t0.div, aes(x=lineage, y=even,color=lineage))+
  geom_boxplot(outlier.shape=NA)+
  scale_colour_manual(values=cols_lineage)+
  geom_jitter(alpha=0.5)+
  xlab("Lineage")+
  ylab("Evenness")+
  theme_bw()+
  #guides(color=guide_legend(title="Reef zone"),shape=guide_legend(title="Reef zone"))+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none")
gg.lin.eve.t0

gg.lin.eve.ps <- ggplot(df.ps.div, aes(x=lineage, y=even,color=lineage))+
  geom_boxplot(outlier.shape=NA)+
  scale_colour_manual(values=cols_lineage)+
  geom_jitter(alpha=0.5)+
  xlab("Lineage")+
  ylab("Evenness")+
  theme_bw()+
  #guides(color=guide_legend(title="Reef zone"),shape=guide_legend(title="Reef zone"))+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none")
gg.lin.eve.ps
```

```{r evenness treatment - prestress only}
#plotting treatment
gg.treat.eve.ps <- ggplot(df.ps.div, aes(x=treat, y=even,color=treat))+
  geom_boxplot(outlier.shape=NA)+
  scale_colour_manual(values=cols_treat_reds)+
  geom_jitter(alpha=0.5)+
  xlab("Treatment")+
  ylab("Evenness")+
  theme_bw()+
  #guides(color=guide_legend(title="Reef zone"),shape=guide_legend(title="Reef zone"))+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none") 
gg.treat.eve.ps

gg.treat.lin.eve.ps <- ggplot(df.ps.div, aes(x=treat, y=even,color=treat))+
  geom_boxplot(outlier.shape=NA)+
  scale_colour_manual(values=cols_treat_reds)+
  geom_jitter(alpha=0.5)+
  xlab("Treatment")+
  ylab("Evenness")+
  theme_bw()+
  #guides(color=guide_legend(title="Reef zone"),shape=guide_legend(title="Reef zone"))+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1), axis.title.x = element_blank(), legend.position = "none")+
  facet_wrap(~lineage)
gg.treat.lin.eve.ps
```

```{r evenness stats}
#evenness
shapiro.test(df.t0.div$even) #nope
leveneTest(df.t0.div$even~lineage,data=df.t0.div) #fine

a.div <- aov(even~lineage,data=df.t0.div)
summary(a.div) 
#             Df Sum Sq  Mean Sq F value Pr(>F)  
# lineage      2 0.0405 0.020240   2.658 0.0806 .
# Residuals   47 0.3579 0.007616                 

a.div <- aov(even~lineage+treat,data=df.ps.div)
summary(a.div) 
#             Df Sum Sq Mean Sq F value Pr(>F)
# lineage      2 0.0488 0.02441   0.789  0.458
# treat        1 0.0522 0.05216   1.686  0.198
# Residuals   85 2.6297 0.03094               

```

## Combine plots so far
```{r evenness stats}
# lineages
plots.lin.t0 = ggarrange(gg.lin.sh.t0, gg.lin.sim.t0, gg.lin.obs.t0, gg.lin.eve.t0, ncol=2, nrow=2)
ggsave(plots.lin.t0, filename = "diversity_plots_lineage_clean_t0.pdf", width=6,height=6, units=c("in"), useDingbats=FALSE)

plots.lin.ps = ggarrange(gg.lin.sh.ps, gg.lin.sim.ps, gg.lin.obs.ps, gg.lin.eve.ps, ncol=2, nrow=2)
ggsave(plots.lin.ps, filename = "diversity_plots_lineage_clean_prestress.pdf", width=6,height=6, units=c("in"), useDingbats=FALSE)

#sites
plots.site.t0 = ggarrange(gg.site.sha.t0, gg.site.sim.t0, gg.site.obs.t0, gg.site.eve.t0, ncol=2, nrow=2)
ggsave(plots.site.t0, filename = "diversity_plots_site_clean_t0.pdf", width=6,height=6, units=c("in"), useDingbats=FALSE)

plots.site.ps = ggarrange(gg.site.sha.ps, gg.site.sim.ps, gg.site.obs.ps, gg.site.eve.ps, ncol=2, nrow=2)
ggsave(plots.site.ps, filename = "diversity_plots_site_clean_prestress.pdf", width=6,height=6, units=c("in"), useDingbats=FALSE)

#treatment
plots.treat.ps = ggarrange(gg.treat.sh.ps, gg.treat.sim.ps, gg.treat.obs.ps, gg.treat.eve.ps, ncol=2, nrow=2)
ggsave(plots.treat.ps, filename = "diversity_plots_treat_clean_prestress.pdf", width=6,height=6, units=c("in"), useDingbats=FALSE)
```

## Hannah stopped here for manuscript. 

# GLMM Mixed Model Analysis
```{r packages for glmm, echo=FALSE}
# following this github pipeline: https://arsweeny.github.io/microbiome-glmm/#2_Mixed_Model_Analysis
library(tidyverse) # data manipulation and ggplot 
library(janitor) # data cleaning 
library(magrittr)  # data manipulation 
library(reshape2) # data manipulation 
library(phyloseq) # output format of sequence processing 
library(microbiome) # useful exploration and visualisation functions 
library(MCMCglmm) # implementation of GLMMs 
#library(ALDEx2) # CLR transformations 
library(plotrix) # std error functions 
library(pals) # palettes 
library(MetBrewer) # palettes 
library(ggregplot) # package my friend made with lots of handy funcions - https://github.com/gfalbery/ggregplot
```

```{r data frame for glmm, echo=FALSE}

psOrder <- tax_glom(ps.clean.t0, taxrank = "Family") # aggregates to the class level 
psMelt <- psmelt(psOrder) # melts into a combined df for plots 
head(psMelt)

SampleDepth <- psMelt %>% 
  group_by(Sample) %>% 
  summarise(Depth=sum(Abundance)) %>% as.data.frame() # summarise reads per sample 
RelAbundance <- full_join(psMelt, SampleDepth) %>% 
  group_by(Sample, Family) %>% 
  summarise(RelAbundance=Abundance/Depth) %>% as.data.frame() # summarise proportion of each phyla for each sample 

require(plotrix)
RelAbundanceMean <- RelAbundance %>% 
  dplyr::select(-Sample) %>%  group_by(Family) %>% 
  summarise_each(funs(mean, sd, std.error)) %>% 
  arrange(desc(mean)) # summarise proportion of phyla across all samples 

RelAbundanceMean
```

```{r actual glmm, echo=FALSE}
str(psMelt)
psMelt2 = psMelt %>%
  select(-Sample_or_Control, -is.neg, -gen_site, -treat, -timepoint) %>%
  mutate_at(vars(lineage, reef, sitename), factor)
str(psMelt2)

# set up model
Mods <- list() # Model list for later 

mf <- 20 # multiplier for model iterations and chains 
resp<-"Abundance" 
fixed<- c("lineage") 
fixed<- c("lineage:sitename") 

mcmcF<- as.formula(paste(resp, "~", paste(fixed)))
  
Prior3 <- list(R = list(V = diag(1), nu = 0.002),
               G = list(G1 = list(V = diag(1), nu = 0.002, alpha.mu = rep(0,1), alpha.V = diag(1)*100),
                        G2 = list(V = diag(1), nu = 0.002, alpha.mu = rep(0,1), alpha.V = diag(1)*100),
                        G3 = list(V = diag(1), nu = 0.002, alpha.mu = rep(0,1), alpha.V = diag(1)*100)))

# to run the model 
Mods[[1]]<- MCMCglmm(fixed= mcmcF, 
                          random= ~ Sample + OTU + lineage:OTU, 
                          prior=Prior3,
                          data=psMelt2,
                          verbose=T, pr=T, pl=T, # will save posteriors for random effect levels & model predictions for each iteration 
                          family = "poisson",
                          nitt = 13000*mf,
                          thin = 10*mf,burnin=3000*mf)

# view proportion of variance associated with each random term
PropVarPois <- MCMCRep(Mods[[1]], scale="link") %>% # function from ggregplot to calculate proportion variance for each random effect 
  as.data.frame() %>% 
  mutate_at(c("Mode", "lHPD", "uHPD"), as.numeric)
PropVarPois

# Visualize variance breakdown for the model
PropVarPois %>% pull(Mode) %>% sum -> TotalVar 
1- TotalVar -> PoissonVar 

PropVarPois[5, ] <- c("poisson", PoissonVar, NA, NA) 

PropVarPois %<>% 
  mutate_at(c("Mode", "lHPD", "uHPD"), as.numeric)

# color scheme
MicroColours <- met.brewer("Renoir",12, "discrete")

PropVarPois%>% 
  mutate(Component=fct_relevel(Component, "Sample", "OTU", "lineage:OTU","units", "poisson")) %>% 
  ggplot(aes(x=1, y=as.numeric(Mode), fill=Component)) + 
  geom_bar(stat="identity", colour="transparent") + 
  scale_x_continuous(labels=NULL) + 
  scale_fill_manual(values=MicroColours[c(2,4,6,8,10)]) +
  theme_bw(base_size = 14) + 
  #theme(legend.position = "top") + 
  theme(axis.ticks.x = element_blank(), 
        legend.text = element_text(size=11), 
        legend.title = element_text(size=12)) + 
  labs(x='Poisson model', y='Proportion Variance')

```





